src/commands/functions/create.js

```
@@ -196,30 +195,78 @@ const pickTemplate = async function () {
 
 const DEFAULT_PRIORITY = 999
 
-/* get functions dir (and make it if necessary) */
-const ensureFunctionDirExists = function (context, flags, config) {
-  const functionsDir = config.build && config.build.functions
-  if (!functionsDir) {
-    context.log(`${NETLIFYDEVLOG} No functions folder specified in netlify.toml`)
-    process.exit(1)
+/**
+ * Get functions directory (and make it if necessary)
+ * @param {FunctionsCreateCommand} context
+ * @returns {string | never} - functions directory or throws an error
+ */
+const ensureFunctionDirExists = async function (context) {
+  const { api, config, site } = context.netlify
+  const siteId = site.id
+  const { log } = context
+  let functionsDirHolder = config.build && config.build.functions
+
+  if (!functionsDirHolder) {
+    log(`${NETLIFYDEVLOG} functions directory not specified in netlify.toml or UI settings`)
+
+    if (!siteId) {
+      context.log(
+        `${NETLIFYDEVERR} Unable to find a site id. Please make sure you are in a directory that is linked to a site with Netlify.`,
+      )
+      process.exit(1)
```

Wasn't totally sure if I should throw an error or use `NETLIFYDEVERR`